tests/runtime/lists/host.ts

```diff
@@ -11,7 +11,7 @@ async function run() {
     testwasi: helpers,
     imports: {
       emptyListParam(a) {
-        assert.deepStrictEqual(Array.from(a), []);
+        assert.deepStrictEqual(Array.from(a as Uint8Array), []);
```

Comment:
This is quite an important consideration actually - basically because "inputs" are from users, we have a more relaxed definition for _lowered input types_ in JavaScript than we do for _lifted output types_. I think we need to more carefully take this into account in the generation process, so that we could relax the input type to `ArrayBuffer | Uint8Array` but at the same type guarantee that the output type is always `Uint8Array`. Any pointers on setting up such a structure welcome, but there were no easy references to use and I didn't want to make timing assumptions.

Reply:
I think this could be solved by threading through a `Mode` parameter or something like that throughout `print_ty`, where it's either `Mode::Lower` or `Mode::Lift` which would enable the ability to print different types.

Reply:
One problem with this approach I believe is that defined types won't get the distinction.